shell/run.ts

```diff
@@ -0,0 +1,181 @@
+ * just the string output.
+ *
+ * @param command The {@link ProcessCommand} to be run
+ * @param options The {@link FullUserRunOptions} governing the behavior of the process runner
+ * @returns A {@link Promise<PipedProcessResult>} containing the result state and output when the process completes
+ */
+$.piped = runPiped;
+
+/**
+ * Runs a process using the streamed output API.
+ *
+ * This means that output will be sent to the same `stdout`/`stderr` as the Deno script, so it will show up in along with
+ * console output.
+ *
+ * @param command The {@link ProcessCommand} to be run
+ * @param options The {@link FullUserRunOptions} governing the behavior of the process runner
+ * @returns A {@link Promise<ProcessResult>} containing the result state when the process completes
+ */
+$.streamed = runStreamed;
+
+/**
+ * Runs a process and checks to see if it was successful.
+ *
+ * @param command The {@link ProcessCommand} to be run
+ * @param options The {@link FullUserRunOptions} governing the behavior of the process runner
+ * @returns A {@link Promise<boolean>} that will resolve to `true` if the process had a zero exit code to indicate success
+ */
+$.try = tryRun;
+
+async function tryRun(
+  command: string[] | string,
+  options: UserTryOptions = {},
+): Promise<boolean> {
+  try {
+    const result = await runPiped(command, { ...options, throwOnError: false });
+    return result.status.success;
+  } catch (err) {
+    if (err instanceof Deno.errors.NotFound) {
+      return false;
+    } else {
+      throw err;
+    }
+  }
+}
+
+function runStreamed(
+  command: string[] | string,
+  options: UserRunOptions = {},
+): Promise<ProcessResult> {
+  const fullOptions: FullRunOptions = {
+    cmd: buildCommandArray(command),
+    stdout: 'inherit',
+    stderr: 'inherit',
+    ...options,
+  };
+  return runInternal(fullOptions);
+}
+
+function runPiped(
+  command: string[] | string,
+  options: UserRunOptions = {},
+): Promise<PipedProcessResult> {
+  const fullOptions: PipedRunOptions = {
+    cmd: buildCommandArray(command),
+    stdout: 'piped',
+    stderr: 'piped',
+    ...options,
+  };
+  return runInternal(fullOptions);
+}
+
+function runGeneric(
+  command: string[] | string,
+  options: FullUserRunOptions = {},
+): Promise<ProcessResult> {
+  const fullOptions: FullRunOptions = {
+    cmd: buildCommandArray(command),
+    ...options,
+  };
+  return runInternal(fullOptions);
+}
+
+function runInternal(options: PipedRunOptions): Promise<PipedProcessResult>;
+function runInternal(options: FullRunOptions): Promise<ProcessResult>;
+async function runInternal(options: FullRunOptions): Promise<ProcessResult> {
+  const proc = Deno.run(options);
+
+  const isPiped = options.stdout === 'piped';
+  const stdoutPromise = isPiped ? proc.output() : Promise.resolve(undefined);
+  const stderrPromise = isPiped ? proc.stderrOutput() : Promise.resolve(undefined);
```

Comment:
Nit - in the functions above you set `options.stderr` values but never use them anywhere.

Reply:
Perhaps they are just used by `Deno.run()` but I would expect it to be respected on this line as well?

Reply:
Done.

The `options.stderr` value is indeed used by `Deno.run()`. The intention was to narrow the API so that stdout and stderr behave the same way (both streamed, or both piped) since configuring them differently seems like a very unlikely thing to ever do on purpose. But while the narrowing done in the `$.piped` and `$.streamed` APIs makes sense, forcing it onto the other functions does indeed feel weird.